jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/internal/ByteBufferChunk.java

```diff
@@ -102,11 +107,40 @@ public ReleasedByConsumer(ByteBuffer byteBuffer, boolean last, Consumer<ByteBuff
             this.releaser = new AtomicReference<>(releaser);
         }
 
-        public void release()
+        @Override
+        public boolean release()
+        {
+            boolean released = super.release();
```

Comment:
This is a wrong implementation. A call to `retain` is a noop and thus a subsequent call to `release` will always release regardless of how many times retain has been called.  This violates the `Retainable` contract.

Do we need this?

Reply:
This should be solved by having `retain()` throw and `release()` returning `true`.